packages/@aws-cdk/aws-applicationautoscaling/lib/schedule.ts

```diff
@@ -17,11 +17,11 @@ export abstract class Schedule {
    * Construct a schedule from an interval and a time unit
    */
   public static rate(duration: Duration): Schedule {
-    const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
-    if (!validDurationUnit.includes(duration.unitLabel())) {
-      throw new Error("Allowed unit for scheduling is: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
-    }
     if (duration.isUnresolved()) {
+      const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
+      if (!validDurationUnit.includes(duration.unitLabel())) {
+        throw new Error("Allowed units for scheduling are: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
+      }
       return new LiteralSchedule(`rate(${duration.formatTokenToNumber()})`);
     }
```

Comment:
```suggestion
    if (duration.isUnresolved()) {
          return new LiteralSchedule(`rate(${duration.formatTokenToNumber()})`);
    }
    
      const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
      if (!validDurationUnit.includes(duration.unitLabel())) {
        throw new Error("Allowed units for scheduling are: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
      }
```

Reply:
GitHub is being weird, so I'll just comment here. It seems like there is still a problem in code, if duration is unresolved the validation will still happen?

Reply:
> if duration is unresolved the validation will still happen?

Yes, that's the intended change here. If duration is a token, it is not possible to perform unit conversion. So the original units specified during construction of the `Duration` object will be used. However, AWS events has a fixed set of units and hence the validation is required.

If Duration is not a token, then the units can be converted from the unsupported set of units to the supported set of units.

Makes sense?

Reply:
Thanks! I missed that `duration.isUnresolved()` is calling the `isUnresolved` method from the `Duration` class rather than the `Token.isUnresolved` method